### PR TITLE
Clean up LiteralField

### PR DESCRIPTION
**Pull request by assertchris**

### CODE REVIEW

File: forms/LiteralField.php [hunk 1]
```diff
@@ -1,76 +1,103 @@
...
- * 
+ *
  * @package forms
  * @subpackage fields-dataless
  */
 class LiteralField extends DatalessField {
-	
 	/**
-	 * @var string $content
+	 * @var string|FormField
```
Comment (tractorcow) [inline forms/LiteralField.php]: I think we are just meant to use string. If you use a FormField as a value, the field internally relies on the `__toString` to cast it safely to HTML safe text. Should we thus limit our expectations to strings only?

Comment (tractorcow) [inline forms/LiteralField.php]: When does this object contain another sub-formfield?

Comment (assertchris) [inline forms/LiteralField.php]: I don't know - Just documenting the behaviour that exists.

Comment (tractorcow) [inline forms/LiteralField.php]: Where did you see this?

Comment (assertchris) [inline forms/LiteralField.php]: [here](https://github.com/silverstripe/silverstripe-framework/pull/4118/files#diff-3f1ed6b2e8300f694532ef48e723667bL35)

Comment (tractorcow) [inline forms/LiteralField.php]: Oh well, merge it now mark as deprecated later I guess.

File: forms/LiteralField.php [hunk 2]
```diff
@@ -1,76 +1,103 @@
...
 	public function FieldHolder($properties = array()) {
-		if(is_object($this->content)) {
-			$obj = $this->content;
-			if($properties)
-				$obj = $obj->customise($properties);
-			return $obj->forTemplate();
-		} else {
-			return $this->content;
+		if($this->content instanceof FormField) {
+			$context = $this->content;
```
Comment (tractorcow) [inline forms/LiteralField.php]: Could the `$this->context` possibly be intended for DBField as well as FormField? Maybe it should be `instanceof ViewableData`? (Although even that method doesn't implement forTemplate, it's the highest common denominator of classes that do).

**Comment by assertchris** [comment]
@tractorcow ping

**Comment by tractorcow** [comment]
It's like, uh, a kind of, literal field, man.

field

**Pull request merged**